File: 83.php
<?php

/**
 * Definition for a singly-linked list.
 * class ListNode {
 *     public $val = 0;
 *     public $next = null;
 *     function __construct($val) { $this->val = $val; }
 * }
 */
class Solution {

    /**
     * @param ListNode $head
     * @return ListNode
     */
    function deleteDuplicates($head) {
        $cur = $head;
        while ($cur && $cur->next) {
            if ($cur->val == $cur->next->val) {
                $cur->next = $cur->next->next;
            } else {
                $cur = $cur->next;
            }
        }
        return $head;
    }
}

File: go/linkedList/83.go
package main

//func deleteDuplicates(head *ListNode) *ListNode {
//	flag := map[int]bool{}
//
//	dummy := new(ListNode)
//	dummy.Next = head
//	cur := dummy
//
//	for cur.Next != nil {
//		if _, ok := flag[cur.Next.Val]; ok {
//			cur.Next = cur.Next.Next
//		} else {
//			flag[cur.Next.Val] = true
//			cur = cur.Next
//		}
//	}
//
//	return dummy.Next
//}
//
//type ListNode struct {
//	Val  int
//	Next *ListNode
//}
